zetup.py

```diff
@@ -0,0 +1,159 @@
+import ziggy.tasks
+import ziggy.shell
+
+
+def hook_on_module_load(project):
+    base_path = os.path.join(project.config['workspace_path'], project.config['root_projects']['flex-fuse']['path'])
+
+    flex_fuse_config = dict()
+    flex_fuse_config['repositories'] = {
+        'flex-fuse': project.config['root_projects']['flex-fuse']
+    }
+    flex_fuse_config['base_path'] = base_path
+    flex_fuse_config['flex_fuse_path'] = os.path.join(base_path, 'flex-fuse')
+    flex_fuse_config['zetup_path'] = os.path.join(flex_fuse_config['flex_fuse_path'], 'zetup.py')
+    flex_fuse_config['zetup_md5'] = ziggy.fs.calculate_file_md5(project.ctx, flex_fuse_config['zetup_path'])
+
+    project.config['flex-fuse'] = flex_fuse_config
+
+
+@defer.inlineCallbacks
+def task_wait_all_projects_updated(project):
+    yield ziggy.tasks.wait_all_projects_updated(project)
+
+
+@defer.inlineCallbacks
+def task_declare_project_updated(project, zetup_path, old_zetup_md5):
+    yield ziggy.tasks.declare_project_updated(project, zetup_path, old_zetup_md5)
+
+
+@defer.inlineCallbacks
+def task_update_sources(project):
+    yield ziggy.tasks.update_sources(project,
+                                     repositories=project.config['flex-fuse']['repositories'],
+                                     base_path=project.config['flex-fuse']['base_path'])
+
+
+@defer.inlineCallbacks
+def task_load_snapshot(project, repo_merges=None):
+    yield ziggy.tasks.load_snapshot(project,
+                                    project_path=project.config['root_projects']['flex-fuse']['path'],
+                                    base_path=project.config['flex-fuse']['base_path'],
+                                    repo_merges=repo_merges)
+
+
+@defer.inlineCallbacks
+def task_take_snapshot(project, filter=None):
+    yield ziggy.tasks.take_snapshot(project,
+                                    repositories=project.config['flex-fuse']['repositories'].keys(),
+                                    project_path=project.config['root_projects']['flex-fuse']['path'],
+                                    base_path=project.config['flex-fuse']['base_path'],
+                                    filter=filter)
+
+
+@defer.inlineCallbacks
+def task_verify_zetup_unchanged(project):
+    yield ziggy.tasks.wait_workspace_updated(project,
+                                             project.config['flex-fuse']['zetup_path'],
+                                             project.config['flex-fuse']['zetup_md5'])
+
+
+@defer.inlineCallbacks
+def task_build_images(project, version, mirror):
+    """
+    Internal build function
+    """
+
+    project.logger.debug('Building', version=version)
+    cwd = project.config['flex-fuse']['flex_fuse_path']
+    cmd = 'make release MIRROR={0} IGUAZIO_VERSION={1}'.format(mirror, version)
+
+    project.logger.debug('Building a release candidate', cwd=cwd, cmd=cmd)
+    out, err, code = yield ziggy.shell.run(project.ctx, cmd, cwd=cwd)
+    if code != 0:
```

Comment:
doesn't shell.run raise an exception on non-zero return?
seems like it: https://github.com/iguazio/ziggy/blob/development/ziggy/shell/__init__.py#L104

so i don't think you need to do this check